design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,515 @@
+
+Please note while `ObserveCreateUpdate` may sound verbose, it accurately
+describes the actions that Crossplane will take on the external resource. This
+naming convention also focuses on what Crossplane will do, rather than what it
+won't do, making it more intuitive and extensible for future policy options,
+such as `ObserveDelete`.
+
+As indicated above, `FullControl` and `ObserveCreateUpdate` policies will behave
+precisely the same as the deletion policies we have today, including keeping the
+default behavior the same. We will introduce the new behavior with the
+`ObserveOnly` option, which would be pretty similar to what we have today to
+[import existing managed resources], but instead of starting to manage after
+import, we will not make any modifications to the external resource and only
+sync status back.
+
+```yaml
+apiVersion: ec2.aws.crossplane.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    crossplane.io/external-name: vpc-12345678
+  name: observe-vpc
+spec:
+  managementPolicy: ObserveOnly
+  forProvider:
+    region: us-east-1
+```
+
+### Implementation
+
+Crossplane providers already manage external resources by implementing the
+Crossplane runtime's ExternalClient interface, which includes the four methods
+listed below.
+
+```go
+type ExternalClient interface {
+	Observe(ctx context.Context, mg resource.Managed) (ExternalObservation, error)
+	Create(ctx context.Context, mg resource.Managed) (ExternalCreation, error)
+	Update(ctx context.Context, mg resource.Managed) (ExternalUpdate, error)
+	Delete(ctx context.Context, mg resource.Managed) error
+}
+```
+
+We will leverage the fact that we have an already implemented Observe method for
+all managed resources by calling only it when the Management Policy is set to
+`ObserveOnly`. This will require minor modifications in the Managed Reconciler
+code that will return early in the reconcile loop and prevent invocation of the
+other methods, namely, Create, Update and Delete. These modifications will
+implement the following logic at a high level:
+
+Right after the `Observe` method invocation, if `ObserveOnly`:
+
+- Return error if the resource does not exist
+- Publish connection details
+- Call `client.Update` if the resource was late initialized
+- Report success and return early
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this
+new policy as an alpha feature that will be off by default and will be
+controlled by `--enable-alpha-management-policies` flag in Providers.
+
+This will not prevent the field from appearing in the schema of the managed
+resources. However, we will ignore the `spec.managementPolicy` when the feature
+is not enabled.
+
+#### Deprecation of `deletionPolicy`
+
+With the new `managementPolicy` covering the existing `deletionPolicy`, we will
+deprecate the latter in favor of the former.
+
+Until we drop the `deletionPolicy` from the schema altogether, we need to be
```

Comment:
I have a question about the future, technically out of scope of your proposal. You mention eventually dropping `deletionPolicy` from the schema. How will that affect customers who are updating to a new version of Crossplane? Will there be mechanism to help them migrate, or do they have to manually migrate from `deletionPolicy` to `managementPolicy`?

Reply:
We basically can't drop it from any `v1beta1` resource. We could introduce `v1` (or `v1beta2`) and drop it there, then after a sufficient amount of time remove the beta APIs altogether.